File: apps/mobile/src/modules/clients/types/company.ts
export type Company = {
	id: string;
	legal_name: string;
	trade_name: string;
	cnpj: string;
	is_active: boolean;
	industry: string;
	phone: string;
	email: string;
	zip_code: string;
	street: string;
	number: string;
	complement: string | null;
	neighborhood: string;
	city: string;
	state: string;
	created_at: string;
	updated_at: string;
};

export type CompanyCreateInput = {
	legal_name: string;
	trade_name: string;
	cnpj: string;
	industry: string;
	phone: string;
	email: string;
	zip_code: string;
	street: string;
	number: string;
	complement?: string;
	neighborhood: string;
	city: string;
	state: string;
};

export type CompanyUpdateInput = CompanyCreateInput & {
	is_active: boolean;
};

File: apps/server/app/modules/clients/models/company.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, String, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.shared.database import Base


class Company(Base):
    __tablename__ = "company"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        primary_key=True,
        default=uuid.uuid4,
    )

    legal_name: Mapped[str] = mapped_column(String(150), nullable=False)
    trade_name: Mapped[str] = mapped_column(String(150), nullable=False)
    cnpj: Mapped[str] = mapped_column(String(18), unique=True, nullable=False)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    industry: Mapped[str] = mapped_column(String(100), nullable=False)
    product: Mapped[str | None] = mapped_column(String(100), nullable=True)

    phone: Mapped[str] = mapped_column(String(30), nullable=False)
    email: Mapped[str] = mapped_column(String(255), unique=True, nullable=False)

    zip_code: Mapped[str] = mapped_column(String(10), nullable=False)
    street: Mapped[str] = mapped_column(String(150), nullable=False)
    number: Mapped[str] = mapped_column(String(10), nullable=False)
    complement: Mapped[str | None] = mapped_column(String(100), nullable=True)
    neighborhood: Mapped[str] = mapped_column(String(100), nullable=False)
    city: Mapped[str] = mapped_column(String(100), nullable=False)
    state: Mapped[str] = mapped_column(String(2), nullable=False)

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )
